### ISSUE TITLE
Az.Automation - Runbookname pattern

### ISSUE THREAD

**Issue by plovegro** [Azure/azure-powershell#20614]
### Description

On the latest version 1.9.0 i am attempting to use Register-AzAutomationScheduledRunbook, which is now throwing an error on my Runbook Name, which previously worked in 1.8.0

My runbook name is "Rebuild Nightly Jobs" but the error i am receiving is this - 

```
Register-AzAutomationScheduledRunbook: /home/azureuser/myagent/_work/
Line |
  10 |  Register-AzAutomationScheduledRunbook -AutomationAccountName "Test-de …
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | 'runbookName' does not match expected pattern
     | '^[a-zA-Z]*-*[a-zA-Z0-9]*$'.
##[error]PowerShell exited with code '1'.
```

I can't see any documentation about this being a breaking change.

### Issue script & Debug output

```PowerShell
Register-AzAutomationScheduledRunbook: /home/azureuser/myagent/_work/
Line |
  10 |  Register-AzAutomationScheduledRunbook -AutomationAccountName "test-de …
     |  ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | 'runbookName' does not match expected pattern
     | '^[a-zA-Z]*-*[a-zA-Z0-9]*$'.
##[error]PowerShell exited with code '1'.
```

### Environment data

```PowerShell
Task         : Azure PowerShell
Description  : Run a PowerShell script within an Azure environment
Version      : 5.209.0
```

### Module versions

```PowerShell
Az.Automation 1.9.0
```

**Comment by dingmeng-xue** [comment]
@plovegro , could you share debug log when you try to create that runbook? You can turn on debug via $DebugPreference = "Continue".

**Comment by plovegro** [comment]
Well the runbook execution was created fine, and it was actually created via a Bicep. Its the scheduling call that is throwing the error - Register-AzAutomationScheduledRunbook

**Comment by navba-MSFT** [comment]
@plovegro Thanks for reaching out to us and reporting this issue. Could you please run `$DebugPreference = "Continue"` and then run your above PS cmdlet and share the debug output here ? Awaiting your reply.

**Comment by plovegro** [comment]
`DEBUG: 2:54:28 PM - RegisterAzureAutomationScheduledRunbook begin processing with ParameterSet 'ByRunbookNameAndScheduleName'.
DEBUG: 2:54:28 PM - using account id ''...
DEBUG: 2:54:28 PM - [ConfigManager] Got nothing from [DisplayBreakingChangeWarning], Module = [], Cmdlet = []. Returning default value [True].
DEBUG: [Common.Authentication]: Authenticating using Account: '', environment: 'AzureCloud', tenant: ''
DEBUG: 2:54:28 PM - [SilentAuthenticator] Calling SharedTokenCacheCredential.GetTokenAsync - TenantId:'',
Scopes:'https://management.core.windows.net//.default', AuthorityHost:'https://login.microsoftonline.com/', UserId:''
DEBUG: SharedTokenCacheCredential.GetToken invoked. Scopes: [ https://management.core.windows.net//.default ] ParentRequestId:
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z -] IsLegacyAdalCacheEnabled: yes
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z -] [Region discovery] Not using a regional authority.
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z -] [Region discovery] Not using a regional authority.
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z -] [Region discovery] Not using a regional authority.
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z - ] IsLegacyAdalCacheEnabled: yes
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z -] IsLegacyAdalCacheEnabled: yes
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z] [WamBroker] WAM supported OS.
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z] [WamBroker] ListWindowsWorkAndSchoolAccounts option was not enabled.
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z - ] [Region discovery] Not using a regional authority.
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z] Found 2 cache accounts and 0 broker accounts
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z] Returning 2 accounts
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z - ] MSAL MSAL.Desktop with assembly version '4.49.1.0'.
 CorrelationId()
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z - ] === AcquireTokenSilent Parameters ===
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z -] LoginHint provided: False
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z - ] Account provided: True
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z -] ForceRefresh: False
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z - 
=== Request Data ===
Authority Provided? - True
Scopes - https://management.core.windows.net//.default
Extra Query Params Keys (space separated) -
ApiId - AcquireTokenSilent
IsConfidentialClient - False
SendX5C - False
LoginHint ? False
IsBrokerConfigured - True
HomeAccountId - False
CorrelationId -
UserAssertion set: False
LongRunningOboCacheKey set: False
Region configured:
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z -] === Token Acquisition (SilentRequest) started:
  Scopes: https://management.core.windows.net//.default
 Authority Host: login.microsoftonline.com
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z - 2] [Region discovery] Not using a regional authority.
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z - ] Access token is not expired. Returning the found
cache entry. [Current time (01/12/2023 03:54:28) - Expiration Time (01/12/2023 05:03:06 +00:00) - Extended Expiration Time (01/12/2023 05:03:06 +00:00)]
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z - ] Returning access token found in cache. RefreshOn
exists ? False
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z -] [Region discovery] Not using a regional authority.
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z - ]
 === Token Acquisition finished successfully:
DEBUG: False MSAL 4.49.1.0 MSAL.Desktop 4.8 or later Windows 10 Enterprise [2023-01-12 03:54:28Z - ]  AT expiration time: 12/01/2023 5:03:06 AM +00:00,
scopes: https://management.core.windows.net//user_impersonation https://management.core.windows.net//.default. source: Cache
DEBUG: SharedTokenCacheCredential.GetToken succeeded. Scopes: [ https://management.core.windows.net//.default ] ParentRequestId:  ExpiresOn: 2023-01-12T05:03:06.0000000+00:00
DEBUG: [Common.Authentication]: Received token with LoginType 'User', Tenant: '', UserId: ''
DEBUG: 2:54:28 PM - [ConfigManager] Got nothing from [EnableDataCollection], Module = [], Cmdlet = []. Returning default value [True].
Register-AzAutomationScheduledRunbook : 'runbookName' does not match expected pattern '^[a-zA-Z]*-*[a-zA-Z0-9]*$'.
At line:1 char:1
+ Register-AzAutomationScheduledRunbook -AutomationAccountName "bt-de ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : CloseError: (:) [Register-AzAutomationScheduledRunbook], ValidationException
    + FullyQualifiedErrorId : Microsoft.Azure.Commands.Automation.Cmdlet.RegisterAzureAutomationScheduledRunbook

DEBUG: 2:54:30 PM - [ConfigManager] Got nothing from [DisplayBreakingChangeWarning], Module = [], Cmdlet = []. Returning default value [True].
DEBUG: AzureQoSEvent:  Module: Az.Automation:1.9.0; CommandName: Register-AzAutomationScheduledRunbook; PSVersion: 5.1.19041.2364; IsSuccess: False; Duration: 00:00:01.3425099; Exception:
'runbookName' does not match expected pattern '^[a-zA-Z]*-*[a-zA-Z0-9]*$'.;
DEBUG: 2:54:30 PM - RegisterAzureAutomationScheduledRunbook end processing.`

**Comment by navba-MSFT** [comment]
@plovegro Thanks for sharing the details. We are looking into this. We will get back to you once we have an update.

**Comment by navba-MSFT** [comment]
@plovegro This issue is by design and not specific to this cmdlet `Register-AzAutomationScheduledRunbook` only.
See [this](https://learn.microsoft.com/en-us/azure/automation/manage-runbooks#:~:text=The%20runbook%20name%20must%20start%20with%20a%20letter%20and%20can%20contain%20letters%2C%20numbers%2C%20underscores%2C%20and%20dashes) documentation. which mentions that `A runbook name can contain only letters, numbers, underscores and dashes, and must begin with a letter.` The same behavior is observed in portal too, see below:

![image](https://user-images.githubusercontent.com/57353862/211999858-911ab6d0-5610-4848-8fe1-c004f9e67acc.png)

**Comment by plovegro** [comment]
Thank you - although that is not quite right. The portal requires only letters, numbers, underscores and dashes. So in theory i could have this - "This-is-my-runbook".
The cmdlet `Register-AzAutomationScheduledRunbook ` requires the specific pattern of - "My-Runbook". 2 words, separated by a hyphen. This is different behaviour to the portal and very confusing.

**Comment by nowakma05** [comment]
Hey all,
@navba-MSFT  Got similar issue with az.automation. I'm receiving same error on execution of `Start-AzAutomationRunbook` 
```
runbookName' does not match expected pattern
     | '^[a-zA-Z]*-*[a-zA-Z0-9]*$'.
##[error]PowerShell exited with code '1'.
```
but name of my runbook is `start-my-runbook` wich is not match the regex in Az.automation v. 1.9.0, but in version 1.8.0 all is working. I could find any info about that change.

**Comment by navba-MSFT** [comment]
@plovegro @novatelli Thanks for your reply. I am looking into this.

**Comment by novatelli** [comment]
@navba-MSFT Hey, is there any update on this?

**Comment by navba-MSFT** [comment]
@novatelli @plovegro I have reached out to the Automation Product Group team and awaiting their reply now.

**Comment by obesser** [comment]
I also get this error with Start-AzAutomationRunbook for runbooks that contain only letters and underscores (allowed in Azure Portal).

**Comment by navba-MSFT** [comment]
@plovegro @novatelli @obesser I haven't heard back from the automation Product Group team yet. I am following up on this with them. I will keep you posted.

**Comment by MainDBA** [comment]
MyAZ.Automation module was updated on 16/01/2023 to version 1.9.0 and I have started to get this error using Start-AzAutomationRunbook to call runbooks whose names contain only letters and underscores.  I have also tested and am able to create a new runbook using letters and underscores.

**Comment by navba-MSFT** [comment]
@plovegro @novatelli @obesser @MainDBA We appreciate your patience on this. I have heard back from the Product Owners. The Automation Dev team is aware of this issue now and they are releasing a fix soon. Until then the workaround is to use the older version ([1.8 ](https://www.powershellgallery.com/packages/Az.Automation/1.8.0)and [below](https://www.powershellgallery.com/packages/Az.Automation/1.7.4)) of the `Az.Automation` module. I will keep you posted about the progress. We sincerely apologize for the inconvenience caused.

**Comment by navba-MSFT** [comment]
@plovegro @novatelli @obesser @MainDBA I am yet to hear back from the Automation Product Owners on the fix release. You could also send an email to me on (navba [@] microsoft .(dot)com ) to stay connected about the updates and fix.

**Comment by peruzzof** [comment]
Hi, I am also impacted by this change, applied the workfix and would like to be informed when the fix is available.

**Comment by thgauchet** [comment]
Hi,
we I am also impacted by this change. Please keep us informed.

**Comment by navba-MSFT** [comment]
@thgauchet Thanks for sending me an email. I will keep you posted about the fix release. 

@peruzzof Could you please send an email to me on (navba [@] microsoft .(dot)com ) to stay connected about the updates and fix ?

**Comment by navba-MSFT** [comment]
cc @jaspkaur28 for awareness.

**Comment by navba-MSFT** [comment]
@plovegro @novatelli @obesser @MainDBA @rednender The Automation Product Group team has been made aware of this issue. They have a PR in place which is currently waiting on the approval. We are following up closely and will update once the changes are checked in. I will keep you all posted about the progress. We sincerely apologize for the inconvenience caused. 

In the meantime, to unblock you the only workaround is to to use the older version ([1.8 ](https://www.powershellgallery.com/packages/Az.Automation/1.8.0)and [below](https://www.powershellgallery.com/packages/Az.Automation/1.7.4)) of the Az.Automation module.

**Comment by navba-MSFT** [comment]
@plovegro @novatelli @obesser @MainDBA @rednender  here is a quick update. The fix is merged and we are working with Azure Powershell team to release it quickly. The tentative ETA is 2/20 .

**Comment by SnhuSpooner** [comment]
@navba-MSFT Thank you for the update.  I too am running into this issue using Microsoft-hosted Agents in Azure DevOps.

Two weeks is a long time to wait.  In the meantime, the Microsoft-hosted Agents should be updated to use Powershell Module Az version 9.2.0 instead of version 9.3.0 so that we no longer experience that issue.  

Is there any reason why that change cannot be made **today** until the fix has been released?

**Comment by peruzzof** [comment]
@SnhuSpooner , there is another team responsible for the Microsoft hosted agents and I believe that there is a process to deploy a new version. I faced the same situation as you and I solved with one script to install the "outdated version" (Install-Module -Name Az.Automation -RequiredVersion 1.8.0 -Force) and, on each script, I had to import the right version (Import-Module Az.Automation -RequiredVersion 1.8.0).
I hope that this help you to recover your pipeline.

Based on the time that I took to update to version 1.9.0 I am assuming that we will must to wait, at least, 2 additional weeks to the updated Az.Automation version be available. So in my team we will evaluate mid-march if we can remove the "workaround".

**Comment by teqqra** [comment]
same issue, starting runbook via invoke from automation account v 1.9.0 dont work,
runbookName' does not match expected pattern
     | '^[a-zA-Z]*-*[a-zA-Z0-9]*$'.

but 1.7.0 does.

Let me know if you need any more info
Thanks!

**Comment by SnhuSpooner** [comment]
@peruzzof I understand there are temporary workarounds that all of us could put in place, whether it is adding a script to set the version or change all runbooks names and the scripts that call them.

However my question is why is Microsoft imposing that action on all of its customers who use Microsoft-hosted Agents?  Microsoft can very simply make one change to set the Az version on the agents they manage; rather than thousands of customers having to spend time and money to create temporary workarounds, and later spend more time and money later when the fix is released to go back an undo all of those temporary workarounds.  Seems like Microsoft doesn't understand the principles of cloud hosted environments and are stuck in a 80's-mindset that customers have to implement workarounds for escaped bugs.

**Comment by navba-MSFT** [comment]
@SnhuSpooner We hear you and understand your concern. So we are working on priority to address this. 
@plovegro @novatelli @obesser @MainDBA @rednender FYI

Here is the newly available PS Az.Automation module: https://www.powershellgallery.com/packages/Az.Automation/1.9.1 Could you please test the same and check if this works ? Awaiting your reply.

**Comment by cameronkaz** [comment]
Cant be tested in Azure DevOps as adding the commands to a Powershell task;
`Get-Module -Name "Az*"
Uninstall-Module Az.Accounts -RequiredVersion 2.11.1 -Force
Install-Module -Name Az.Accounts -Force -MinimumVersion 2.11.2
Import-Module -Name Az.Accounts -Force -MinimumVersion 2.11.2
Install-Module -Name Az.Automation -Force -MinimumVersion 1.9.1
Import-Module -Name Az.Automation -MinimumVersion 1.9.1
Get-Module -Name "Az*"`

Does not work, may work when manually executing, but this is not how I use this module. Please update as required.

**Comment by navba-MSFT** [comment]
@cameronkaz The hotfix Az.Automation v1.9.1 is an (OOB) out-of-band release. It is NOT included in the current Az v9.4.0, but it will be included in the next release of Az.

For now, Az.Automation v1.9.1 needs to be installed separately from Az v9.4.0 since it is an OOB. So, unfortunately there is no ETA that is shared on when Microsoft-hosted Agent VM images that includes automation bug fix. 

In meanwhile, as a workaround, you can use either use Self-Hosted agents (or) change the image configuration of Microsoft-Hosted Image to add preferred Azure PowerShell version.

<img width="376" alt="image" src="https://user-images.githubusercontent.com/57353862/218952522-27a4c2ab-b594-4238-8eca-1fafd23991c2.png">

**Comment by jformacek** [comment]
@navba-MSFT : Confirming that proposed workaround works on MS-hosted agent
